### ISSUE TITLE
Parse MySQL multi-table UPDATE

### ISSUE THREAD

**Issue by jasonaowen** [sqlfluff/sqlfluff#2832]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff is unable to parse [MySQL multiple table `UPDATE` statements](https://dev.mysql.com/doc/refman/5.7/en/update.html).

### Expected Behaviour

SQLFluff to parse the valid, executable SQL and report on any rule violations.

### Observed Behaviour

> Line 1, Position 1: Found unparsable section: 'UPDATE a, b SET a.name = b.name WHERE a....

### How to reproduce

With `multi-update.sql` as
```sql
UPDATE a, b SET a.name = b.name WHERE a.id = b.id;
```

running SQLFluff results in a parse error:

```sh
$ sqlfluff parse --verbose --ignore-local-config --dialect mysql multi-update.sql 
==== sqlfluff ====
sqlfluff:               0.11.0 python:                 3.8.10
implementation:        cpython dialect:                 mysql
verbosity:                   1 templater:               jinja

=== [ path: multi-update.sql ] ===

[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        update_statement:
[L:  1, P:  1]      |            unparsable:                                       !! Expected: 'UpdateStatementSegment'
[L:  1, P:  1]      |                keyword:                                      'UPDATE'
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                raw:                                          'a'
[L:  1, P:  9]      |                comma:                                        ','
[L:  1, P: 10]      |                whitespace:                                   ' '
[L:  1, P: 11]      |                raw:                                          'b'
[L:  1, P: 12]      |                whitespace:                                   ' '
[L:  1, P: 13]      |                raw:                                          'SET'
[L:  1, P: 16]      |                whitespace:                                   ' '
[L:  1, P: 17]      |                raw:                                          'a'
[L:  1, P: 18]      |                raw:                                          '.'
[L:  1, P: 19]      |                raw:                                          'name'
[L:  1, P: 23]      |                whitespace:                                   ' '
[L:  1, P: 24]      |                raw:                                          '='
[L:  1, P: 25]      |                whitespace:                                   ' '
[L:  1, P: 26]      |                raw:                                          'b'
[L:  1, P: 27]      |                raw:                                          '.'
[L:  1, P: 28]      |                raw:                                          'name'
[L:  1, P: 32]      |                whitespace:                                   ' '
[L:  1, P: 33]      |                raw:                                          'WHERE'
[L:  1, P: 38]      |                whitespace:                                   ' '
[L:  1, P: 39]      |                raw:                                          'a'
[L:  1, P: 40]      |                raw:                                          '.'
[L:  1, P: 41]      |                raw:                                          'id'
[L:  1, P: 43]      |                whitespace:                                   ' '
[L:  1, P: 44]      |                raw:                                          '='
[L:  1, P: 45]      |                whitespace:                                   ' '
[L:  1, P: 46]      |                raw:                                          'b'
[L:  1, P: 47]      |                raw:                                          '.'
[L:  1, P: 48]      |                raw:                                          'id'
[L:  1, P: 50]      |    statement_terminator:                                     ';'
[L:  1, P: 51]      |    newline:                                                  '\n'

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'UPDATE a, b SET
                       | a.name = b.name WHERE a....'
```

### Dialect

MySQL

### Version

sqlfluff, version 0.11.0
Python 3.8.10

### Configuration

All defaults

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by jasonaowen** [comment]
This may result in a similar set of changes as for #2819! Unfortunately I don't have a dev environment for SQLFluff set up yet, and I haven't had a chance to study the changes in #2823 deeply enough to replicate them for this similar but distinct context.
